swarm/storage/resource_test.go

```diff
@@ -456,3 +458,99 @@ func (self *testValidator) checkAccess(name string, address common.Address) (boo
 func (self *testValidator) nameHash(name string) common.Hash {
 	return self.hashFunc(name)
 }
+
+func TestResourceMultihash(t *testing.T) {
+
+	// signer containing private key
+	//	signer, err := newTestSigner()
+	//	if err != nil {
+	//		t.Fatal(err)
+	//	}
+
+	// make fake backend, set up rpc and create resourcehandler
+	backend := &fakeBackend{
+		blocknumber: int64(startBlock),
+	}
+
+	// set up rpc and create resourcehandler
+	rh, _, _, teardownTest, err := setupTest(backend, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer teardownTest()
+
+	// create a new resource
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	_, err = rh.NewResource(ctx, safeName, resourceFrequency)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// we're naïvely assuming keccak256 for swarm hashes
+	// if it ever changes this test should also change
```

Comment:
do we need to PR the swarm BMT hash to multihash?

Reply:
I don't know, do we? Let's discuss separately.

Reply:
We should, along with a spec of its exact construction. We currently have [a hack in go-meta](https://github.com/meta-network/go-meta/blob/master/meta.go#L284-L288) to register it ourselves.

Reply:
Let's make a separate PR?